vm/src/pyobject.rs

```diff
@@ -284,46 +283,34 @@ impl PyContext {
     where
         F: IntoPyNativeFunc<FKind>,
     {
-        PyObject::new(
-            PyBuiltinFunction::from(f.into_func()),
-            self.types.builtin_function_or_method_type.clone(),
-            None,
-        )
+        PyFuncDef::from(f.into_func()).build_function(self)
     }
 
-    pub fn new_function_named<F, FKind>(&self, f: F, module: String, name: String) -> PyObjectRef
+    pub(crate) fn new_stringref(&self, s: String) -> objstr::PyStringRef {
```

Comment:
I think new_str can return PyStringRef and PyObjectRef can be returned by vm.new_pyobj

Reply:
I think I'll do that separately; there's a lot of code right now that assumes that new_str will return a PyObjectRef so it'd be better to change it in another PR